modulus/models/module.py

```diff
@@ -65,55 +106,286 @@ def debug(self):
 
+    @classmethod
+    def from_checkpoint(cls, file_name: str) -> None:
+        """Simple utility for constructing a model from a checkpoint
+
+        Parameters
+        ----------
+        file_name : str
+            Checkpoint file name
+
+        Returns
+        -------
+        Module
+
+        Raises
+        ------
+        IOError
+            If file_name provided does not exist or is not a valid checkpoint
+        """
+
+        directory = Module._check_checkpoint(file_name)
+
+        with open(directory.joinpath("args.json"), "r") as f:
+            args = json.load(f)
+
+        model = cls.instantiate(args)
+
+        model_dict = torch.load(
+            directory.joinpath("model.pt"), map_location=model.device
+        )
+        model.load_state_dict(model_dict)
+
+        return model
+
+    @staticmethod
+    def from_torch(torch_model_class, meta=None):
+        """Construct a Modulus module from a PyTorch module
+
+        Parameters
+        ----------
+        torch_model_class : torch.nn.Module
+            PyTorch module class
+        meta : ModelMetaData, optional
+            Meta data for the model, by default None
+
+        Returns
+        -------
+        Module
+        """
+
+        # Define an internal class as before
+        class ModulusModel(Module):
+            def __init__(self, *args, **kwargs):
+                super().__init__(meta=meta)
+                self.inner_model = torch_model_class(*args, **kwargs)
+
+            def forward(self, x):
+                return self.inner_model(x)
+
+        # Get the argument names and default values of the PyTorch model's init method
+        init_argspec = inspect.getfullargspec(torch_model_class.__init__)
+        model_argnames = init_argspec.args[1:]  # Exclude 'self'
+        model_defaults = init_argspec.defaults or []
+        defaults_dict = dict(
+            zip(model_argnames[-len(model_defaults) :], model_defaults)
+        )
+
+        # Define the signature of new init
+        params = [inspect.Parameter("self", inspect.Parameter.POSITIONAL_OR_KEYWORD)]
+        params += [
+            inspect.Parameter(
+                argname,
+                inspect.Parameter.POSITIONAL_OR_KEYWORD,
+                default=defaults_dict.get(argname, inspect.Parameter.empty),
+            )
+            for argname in model_argnames
+        ]
+        init_signature = inspect.Signature(params)
+
+        # Replace ModulusModel.__init__ signature with new init signature
+        ModulusModel.__init__.__signature__ = init_signature
+
+        # Generate a unique name for the created class
+        new_class_name = f"{torch_model_class.__name__}ModulusModel"
+        ModulusModel.__name__ = new_class_name
+
+        # Add this class to the dict of models classes
+        ModulusModel.register(ModulusModel, new_class_name)
+
+        return ModulusModel
+
+    @classmethod
+    def register(cls, model, name: Union[str, None] = None):
```

Comment:
Would prefer the registry to be a separate singleton class rather than a class variable in `Module`. IMO it is a much clearer API to have `modulus.registry.register(ModulusModel, new_class_name)` than to do `ModulusModel.register(ModulusModel, new_class_name)`